docs/file-formats/kmx-file-format.md

```diff
@@ -0,0 +1,301 @@
+# KMX Binary Format
+
+This document describes the binary format for KMX, compiled from Keyman .kmn
+source files.
+
+## Principles
+
+- All integer values are unsigned 32-bit little-endian unless otherwise
+  specified.
+- All offsets are 32-bit little-endian values. All offsets are relative to the
+  beginning of the data (normally start of file). A zero offset indicates that
+  the value is not present, equivalent to `NULL` -- no offset can ever point to
+  the start of the file.
+- All strings are UTF-16LE unless otherwise specified and null terminated.
+  Strings are referenced by offset (see above).
+- The file format has a header section followed by data. There is no specific
+  defined order for data in the file. Empty space is permitted but should be
+  zeroed. Note that `COMP_KEYBOARD_KMXPLUS` must follow immediately after
+  `COMP_KEYBOARD` with no empty space (that is at byte 0x40).
+
+## `COMP_KEYBOARD`: KMX header
+
+The very first section is a header that describes the remainder of the file.
+
+| ∆ | Bits | Name                 | Description                                                                       |
```

Comment:
∆ here is in decimal, not hex... right?

Just checking, since at line 126 we refer to the hex value, not the decimal value.

Reply:
Yes. Agree that the difference is somewhat confusing. The decimal here matches what we did for the kmxplus spec.